Ability to hide output properties with other misc fixes

tomy2105 opened the pull request:
This solves #3022.
In addition it solves  #3247 and #3248 I've discovered while implementing #3022.

I've also added new sample which demonstrate how to add new Activity derived from `HttpEndpoint` activity to act as new trigger activity (with different input and output parameters). This was the use case why I needed to hide output properties in the first place.

I've discovered, in two places, I needed to replace `var activityDescriptor = await activityType.DescribeAsync();` with `var activityDescriptor = await _activityTypeService.DescribeActivityType(activityType);` because the first one doesn't invoke `INotificationHandler<DescribingActivityType>` handlers. Hope both places are ok with that.

sfmskywalker commented on src/samples/aspnet/Elsa.Samples.ExtendHttpEndpoint/Elsa.Samples.ExtendHttpEndpoint.csproj:
```diff
@@ -0,0 +1,26 @@
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <Content Remove="workflow-definition-custom-http-endpoint.json" />
+    </ItemGroup>
```
Can we remove this?

tomy2105 replied:
@sfmskywalker, workflow-definition-custom-http-endpoint.json is an designer export of the simple sample workflow using my "custom" HTTP Endpoint derived activity.
I didn't know where to put it so I placed in the folder of the sample and excluded it from C# project (hence the entry in .csproj).
Do you want me to include it in the project (because that will happen if I remove these 3 lines which exclude it)?
Or you want me to complete remove this json file from the git?

tomy2105 replied:
@sfmskywalker I've removed this json file completely from git and attaching to this PR, this is simplest solution (I wanted to clear this up this week :)).

sfmskywalker reviewed (approved):
Excellent, thanks @tomy2105 ! And additional kudos for the sample 🎉

tomy2105 commented:
Rebased to fix conflicts in sln and removed this file and attached here for reference....
[workflow-definition-custom-http-endpoint.zip](https://github.com/elsa-workflows/elsa-core/files/9381965/workflow-definition-custom-http-endpoint.zip)

sfmskywalker commented:
No, once the merge conflict is resolved we can merge 👍🏻

sjd2021 commented:
Hi, 
Trying to follow the ci/cd flow from this PR so I can pull this version down from nuget. I see 2.9.x associated with this PR while nuget seems to be at 2.8. Is there anything else that needs to happen other than it going to master in order for the latest package ref to be updated there?
